networking/src/p2p/peer.rs

```diff
@@ -636,3 +871,150 @@ async fn begin_process_incoming(
+                }
+                Ok(())
+            }
+        }
+
+        let drain = slog_async::Async::new(slog::Duplicate::new(MyDrain(warns), drain).fuse())
+            .build()
+            .filter_level(level)
+            .fuse();
+
+        Logger::root(drain, slog::o!())
+    }
+
+    fn create_test_actor_system(log: Logger) -> ActorSystem {
+        SystemBuilder::new()
+            .name("create_actor_system")
+            .log(log)
+            .create()
+            .expect("Failed to create test actor system")
+    }
+
+    fn create_test_tokio_runtime() -> tokio::runtime::Runtime {
+        tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .expect("Failed to create test tokio runtime")
+    }
+
+    fn create_test_peer(
+        sys: &impl ActorRefFactory,
+        network_channel: NetworkChannelRef,
+        tokio_executor: Handle,
+        log: Logger,
+    ) -> PeerRef {
+        let node_identity = Arc::new(Identity::generate(0f64).unwrap());
+        let peer_public_key_hash: CryptoboxPublicKeyHash =
+            node_identity.public_key.public_key_hash().unwrap();
+        let peer_id_marker = peer_public_key_hash.to_base58_check();
+
+        Peer::actor(
+            "test-peer",
+            sys,
+            network_channel,
+            tokio_executor,
+            BootstrapOutput(
+                Arc::new(Mutex::new(None)),
+                Arc::new(Mutex::new(None)),
+                peer_public_key_hash,
+                peer_id_marker,
+                MetadataMessage::new(false, false).clone(),
+                NetworkVersion::new("".to_owned(), 0, 0),
+                "127.0.0.1:9732".parse().unwrap(),
+            ),
+            &log,
+        )
+        .expect("Cannot create a test actor")
+    }
+
+    fn create_test_mgs() -> PeerMessageResponse {
+        PeerMessage::Advertise(AdvertiseMessage::new(&[])).into()
+    }
+
+    #[test]
+    #[ignore]
+    fn test_quota_exceeded() {
+        let received_messages = Arc::new(AtomicUsize::new(0));
+        let log = create_logger(received_messages.clone(), Level::Debug);
+        let actor_system = create_test_actor_system(log.clone());
+        let runtime = create_test_tokio_runtime();
+        let network_channel =
+            NetworkChannel::actor(&actor_system).expect("Failed to create network channel");
+        let peer = create_test_peer(
+            &actor_system,
+            network_channel.clone(),
+            runtime.handle().clone(),
+            log.clone(),
+        );
+        let msg = create_test_mgs();
+        let idx = ThrottleQuota::msg_index(&msg);
+        for _ in 0..super::THROTTLING_QUOTA_MAX[idx].0 + 11 {
+            peer.tell(SendMessage::new(Arc::new(msg.clone())), None);
+        }
+        std::thread::sleep(Duration::from_millis(50));
+
+        // 11 warnings on dropped messages
+        assert_eq!(received_messages.load(Ordering::Relaxed), 11);
```

Comment:
Ok, this is related to my comments above, I see that because of how it is tested the warning needs to be logged for every message.

I don't know what is the amount of values expected, but if it is thousands, do we want thousands warnings in the logs?